### PR TITLE
DDP 8421 singular update

### PR DESCRIPTION
**Pull request by ssettipalli**
look at https://broadinstitute.atlassian.net/browse/DDP-8423

Root cause:
Announcements were fine during initial insert of announcements.
As part of patch we are executing UpdateTemplatesInPlace and that caused issue for these Announcements .
UpdateTemplatesInPlace is using event trigger and event order as unique identifier to save events.
Since these Announcement events had same trigger and order, it caused issue and messed up templates.
UpdateTemplatesInPlace deleted and recreated template substitutions incorrectly.
changing the event order to be unique fixes issue.
At some point of time we can revisit and look to make UpdateTemplatesInPlace (event unique key gen) more robust. Created https://broadinstitute.atlassian.net/browse/DDP-8575 to track it.

QA:
Tested by deploying to local pre code fix and post fix to make sure issue is fixed.
 
 
 Output that shows that UpdateTemplatesInPlace deleted and recreated/updated template substitutions
 14:01:56.903  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_PARENTAL/COMPLETE-4 announcementMessageTemplate 1004970] updated template text
14:01:56.908  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_PARENTAL/COMPLETE-4 announcementMessageTemplate 1004970] variable singular_announcement_enroll_child_p1 language en: deleted substitution
14:01:56.910  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_PARENTAL/COMPLETE-4 announcementMessageTemplate 1004970] deleted variable singular_announcement_enroll_child_p1
14:01:56.914  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_PARENTAL/COMPLETE-4 announcementMessageTemplate 1004970] inserted variable singular_announcement_enroll_depend_p1
14:01:56.917  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_PARENTAL/COMPLETE-4 announcementMessageTemplate 1004970] variable singular_announcement_enroll_depend_p1 language en: inserted substitution
14:01:56.937  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_DEPENDENT/COMPLETE-6 announcementMessageTemplate 1004972] updated template text
14:01:56.939  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_DEPENDENT/COMPLETE-6 announcementMessageTemplate 1004972] variable singular_announcement_enroll_depend_p1 language en: deleted substitution
14:01:56.942  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_DEPENDENT/COMPLETE-6 announcementMessageTemplate 1004972] deleted variable singular_announcement_enroll_depend_p1
14:01:56.944  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_DEPENDENT/COMPLETE-6 announcementMessageTemplate 1004972] inserted variable singular_announcement_enroll_child_p1
14:01:56.945  C: S: [main] INFO o.b.d.s.task.UpdateTemplatesInPlace [event ACTIVITY_STATUS/ADD_PARTICIPANT_DEPENDENT/COMPLETE-6 announcementMessageTemplate 1004972] variable singular_announcement_enroll_child_p1 language en: inserted substitution

### CODE REVIEW

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java [hunk 1]
```diff
@@ -190,6 +190,8 @@ private void traverseEventConfigurations(Handle handle, long studyId) {
     }
 
     // Best-effort attempt at making a unique identifier for event configuration.
+    // watch out for scenarios where diff events might end up with same unique identifier
+    // ex:- trigger and order are same but diff pre/cancel expressions !
```
Comment (bskinner) [inline pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java:194]: If the `preconditionExpr` and other expressions result in technically different events, should we be incorporating those into the hash?

**Review by bskinner — COMMENTED** [review]
Looks good to me- just had a question regarding the generation of a unique id.

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java (hunk 1 above)
Comment (bskinner) [reply pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java:194]: Ahh- I see your note in the description.

**Review by bskinner — APPROVED** [review]

File: pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java (hunk 1 above)
Comment (ssettipalli) [reply pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateTemplatesInPlace.java:194]: created tkt to track it https://broadinstitute.atlassian.net/browse/DDP-8575

**Pull request merged**